virttest/qemu_monitor.py

```diff
@@ -803,10 +805,10 @@ def __init__(self, vm, name, monitor_params, suppress_exceptions=False):
                 self.on_str, self.off_str = "on", "off"
 
         except MonitorError as e:
-            self._close_sock()
             if suppress_exceptions:
                 logging.warn(e)
             else:
+                self._close_sock()
```

Comment:
no reference for this change in commit msg, could you pls add the reason for avoid closing the socket.

Reply:
> no reference for this change in commit msg, could you pls add the reason for avoid closing the socket.

I prefer that this is an existing bug, when `suppress_exceptions = True`, close the socket will cause monitor connection failed too, this is not the expect result.   Let me explain this in the commit message.

Reply:
@sathnaga  updated, could you help review again, thanks!